src/vs/workbench/services/configuration/browser/configuration.ts

```diff
@@ -415,6 +415,11 @@ export class WorkspaceConfiguration extends Disposable {
 		this._workspaceConfiguration = this._cachedConfiguration = new CachedWorkspaceConfiguration(configurationCache);
 	}
 
+	public dispose() {
+		super.dispose();
+		this._workspaceConfiguration.dispose();
```

Comment:
I think a `MutableDisposable` would be easier to use here.

Reply:
I agree, also another variant would be to wrap with `this._register` a few lines above in the ctor. Leaving to @sandy081 to decide.

Reply:
Refactored. Please check latest changes.